// brace/ext/py-code.js
const pythonCode = [{
    name: 'microbit', // 提示的名称，和value保持一致
    value: 'microbit', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'panic', // 提示的名称，和value保持一致
      value: 'panic', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'sleep', // 提示的名称，和value保持一致
      value: 'sleep', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'running_time', // 提示的名称，和value保持一致
      value: 'running_time', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'temperature', // 提示的名称，和value保持一致
      value: 'temperature', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'accelerometer', // 提示的名称，和value保持一致
    value: 'accelerometer', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'get_x', // 提示的名称，和value保持一致
      value: 'get_x', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'get_y', // 提示的名称，和value保持一致
      value: 'get_y', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'get_z', // 提示的名称，和value保持一致
      value: 'get_z', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'is_gesture', // 提示的名称，和value保持一致
      value: 'is_gesture', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'was_gesture', // 提示的名称，和value保持一致
      value: 'was_gesture', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'get_gestures', // 提示的名称，和value保持一致
      value: 'get_gestures', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'button_a', // 提示的名称，和value保持一致
    value: 'button_a', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'is_pressed', // 提示的名称，和value保持一致
      value: 'is_pressed', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'was_pressed', // 提示的名称，和value保持一致
      value: 'was_pressed', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'get_presses', // 提示的名称，和value保持一致
      value: 'get_presses', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'button_b', // 提示的名称，和value保持一致
    value: 'button_b', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'is_pressed', // 提示的名称，和value保持一致
      value: 'is_pressed', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'was_pressed', // 提示的名称，和value保持一致
      value: 'was_pressed', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'get_presses', // 提示的名称，和value保持一致
      value: 'get_presses', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'compass', // 提示的名称，和value保持一致
    value: 'compass', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'is_calibrated', // 提示的名称，和value保持一致
      value: 'is_calibrated', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'calibrate', // 提示的名称，和value保持一致
      value: 'calibrate', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'clear_calibration', // 提示的名称，和value保持一致
      value: 'clear_calibration', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'get_x', // 提示的名称，和value保持一致
      value: 'get_x', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'get_y', // 提示的名称，和value保持一致
      value: 'get_y', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'get_z', // 提示的名称，和value保持一致
      value: 'get_z', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'get_field_strength', // 提示的名称，和value保持一致
      value: 'get_field_strength', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'heading', // 提示的名称，和value保持一致
      value: 'heading', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'display', // 提示的名称，和value保持一致
    value: 'display', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'show', // 提示的名称，和value保持一致
      value: 'show', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'scroll', // 提示的名称，和value保持一致
      value: 'scroll', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'clear', // 提示的名称，和value保持一致
      value: 'clear', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'get_pixel', // 提示的名称，和value保持一致
      value: 'get_pixel', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'set_pixel', // 提示的名称，和value保持一致
      value: 'set_pixel', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'on', // 提示的名称，和value保持一致
      value: 'on', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'off', // 提示的名称，和value保持一致
      value: 'off', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'is_on', // 提示的名称，和value保持一致
      value: 'is_on', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'pin0', // 提示的名称，和value保持一致
    value: 'pin0', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'is_touched', // 提示的名称，和value保持一致
      value: 'is_touched', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'read_digital', // 提示的名称，和value保持一致
      value: 'read_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'write_digital', // 提示的名称，和value保持一致
      value: 'write_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'write_analog', // 提示的名称，和value保持一致
      value: 'write_analog', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'set_analog_period', // 提示的名称，和value保持一致
      value: 'set_analog_period', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'set_analog_period_microseconds', // 提示的名称，和value保持一致
      value: 'set_analog_period_microseconds', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'pin1', // 提示的名称，和value保持一致
    value: 'pin1', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'is_touched', // 提示的名称，和value保持一致
      value: 'is_touched', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'read_digital', // 提示的名称，和value保持一致
      value: 'read_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'write_digital', // 提示的名称，和value保持一致
      value: 'write_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'write_analog', // 提示的名称，和value保持一致
      value: 'write_analog', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'set_analog_period', // 提示的名称，和value保持一致
      value: 'set_analog_period', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'set_analog_period_microseconds', // 提示的名称，和value保持一致
      value: 'set_analog_period_microseconds', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'pin2', // 提示的名称，和value保持一致
    value: 'pin2', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'is_touched', // 提示的名称，和value保持一致
      value: 'is_touched', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'read_digital', // 提示的名称，和value保持一致
      value: 'read_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'write_digital', // 提示的名称，和value保持一致
      value: 'write_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'write_analog', // 提示的名称，和value保持一致
      value: 'write_analog', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'set_analog_period', // 提示的名称，和value保持一致
      value: 'set_analog_period', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'set_analog_period_microseconds', // 提示的名称，和value保持一致
      value: 'set_analog_period_microseconds', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'pin3', // 提示的名称，和value保持一致
    value: 'pin3', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'read_digital', // 提示的名称，和value保持一致
      value: 'read_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'write_digital', // 提示的名称，和value保持一致
      value: 'write_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'write_analog', // 提示的名称，和value保持一致
      value: 'write_analog', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'set_analog_period', // 提示的名称，和value保持一致
      value: 'set_analog_period', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'set_analog_period_microseconds', // 提示的名称，和value保持一致
      value: 'set_analog_period_microseconds', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'pin4', // 提示的名称，和value保持一致
    value: 'pin4', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'read_digital', // 提示的名称，和value保持一致
      value: 'read_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'write_digital', // 提示的名称，和value保持一致
      value: 'write_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'write_analog', // 提示的名称，和value保持一致
      value: 'write_analog', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'set_analog_period', // 提示的名称，和value保持一致
      value: 'set_analog_period', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'set_analog_period_microseconds', // 提示的名称，和value保持一致
      value: 'set_analog_period_microseconds', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'pin5', // 提示的名称，和value保持一致
    value: 'pin5', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'read_digital', // 提示的名称，和value保持一致
      value: 'read_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'write_digital', // 提示的名称，和value保持一致
      value: 'write_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'pin6', // 提示的名称，和value保持一致
    value: 'pin6', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'read_digital', // 提示的名称，和value保持一致
      value: 'read_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'write_digital', // 提示的名称，和value保持一致
      value: 'write_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'pin7', // 提示的名称，和value保持一致
    value: 'pin7', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'read_digital', // 提示的名称，和value保持一致
      value: 'read_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'write_digital', // 提示的名称，和value保持一致
      value: 'write_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'pin8', // 提示的名称，和value保持一致
    value: 'pin8', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'read_digital', // 提示的名称，和value保持一致
      value: 'read_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'write_digital', // 提示的名称，和value保持一致
      value: 'write_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'pin9', // 提示的名称，和value保持一致
    value: 'pin9', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'read_digital', // 提示的名称，和value保持一致
      value: 'read_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'write_digital', // 提示的名称，和value保持一致
      value: 'write_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'pin10', // 提示的名称，和value保持一致
    value: 'pin10', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'read_digital', // 提示的名称，和value保持一致
      value: 'read_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'write_digital', // 提示的名称，和value保持一致
      value: 'write_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'write_analog', // 提示的名称，和value保持一致
      value: 'write_analog', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'set_analog_period', // 提示的名称，和value保持一致
      value: 'set_analog_period', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'set_analog_period_microseconds', // 提示的名称，和value保持一致
      value: 'set_analog_period_microseconds', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'pin11', // 提示的名称，和value保持一致
    value: 'pin11', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'read_digital', // 提示的名称，和value保持一致
      value: 'read_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'write_digital', // 提示的名称，和value保持一致
      value: 'write_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'pin12', // 提示的名称，和value保持一致
    value: 'pin12', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'read_digital', // 提示的名称，和value保持一致
      value: 'read_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'write_digital', // 提示的名称，和value保持一致
      value: 'write_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'pin13', // 提示的名称，和value保持一致
    value: 'pin13', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'read_digital', // 提示的名称，和value保持一致
      value: 'read_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'write_digital', // 提示的名称，和value保持一致
      value: 'write_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'pin14', // 提示的名称，和value保持一致
    value: 'pin14', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'read_digital', // 提示的名称，和value保持一致
      value: 'read_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'write_digital', // 提示的名称，和value保持一致
      value: 'write_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'pin15', // 提示的名称，和value保持一致
    value: 'pin15', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'read_digital', // 提示的名称，和value保持一致
      value: 'read_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'write_digital', // 提示的名称，和value保持一致
      value: 'write_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'pin16', // 提示的名称，和value保持一致
    value: 'pin16', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'read_digital', // 提示的名称，和value保持一致
      value: 'read_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'write_digital', // 提示的名称，和value保持一致
      value: 'write_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'pin19', // 提示的名称，和value保持一致
    value: 'pin19', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'read_digital', // 提示的名称，和value保持一致
      value: 'read_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'write_digital', // 提示的名称，和value保持一致
      value: 'write_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'pin20', // 提示的名称，和value保持一致
    value: 'pin20', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'read_digital', // 提示的名称，和value保持一致
      value: 'read_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'write_digital', // 提示的名称，和value保持一致
      value: 'write_digital', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'i2c', // 提示的名称，和value保持一致
    value: 'i2c', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'init', // 提示的名称，和value保持一致
      value: 'init', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'read', // 提示的名称，和value保持一致
      value: 'read', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'write', // 提示的名称，和value保持一致
      value: 'write', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'Image', // 提示的名称，和value保持一致
    value: 'Image', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'width', // 提示的名称，和value保持一致
      value: 'width', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'height', // 提示的名称，和value保持一致
      value: 'height', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'get_pixel', // 提示的名称，和value保持一致
      value: 'get_pixel', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'set_pixel', // 提示的名称，和value保持一致
      value: 'set_pixel', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'shift_left', // 提示的名称，和value保持一致
      value: 'shift_left', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'shift_right', // 提示的名称，和value保持一致
      value: 'shift_right', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'shift_up', // 提示的名称，和value保持一致
      value: 'shift_up', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'shift_down', // 提示的名称，和value保持一致
      value: 'shift_down', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'copy', // 提示的名称，和value保持一致
      value: 'copy', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'crop', // 提示的名称，和value保持一致
      value: 'crop', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'invert', // 提示的名称，和value保持一致
      value: 'invert', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'HEART', // 提示的名称，和value保持一致
      value: 'HEART', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'HEART_SMALL', // 提示的名称，和value保持一致
      value: 'HEART_SMALL', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'HAPPY', // 提示的名称，和value保持一致
      value: 'HAPPY', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'SMILE', // 提示的名称，和value保持一致
      value: 'SMILE', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'SAD', // 提示的名称，和value保持一致
      value: 'SAD', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'CONFUSED', // 提示的名称，和value保持一致
      value: 'CONFUSED', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'ANGRY', // 提示的名称，和value保持一致
      value: 'ANGRY', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'ASLEEP', // 提示的名称，和value保持一致
      value: 'ASLEEP', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'SURPRISED', // 提示的名称，和value保持一致
      value: 'SURPRISED', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'SILLY', // 提示的名称，和value保持一致
      value: 'SILLY', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'FABULOUS', // 提示的名称，和value保持一致
      value: 'FABULOUS', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'MEH', // 提示的名称，和value保持一致
      value: 'MEH', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'YES', // 提示的名称，和value保持一致
      value: 'YES', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'NO', // 提示的名称，和value保持一致
      value: 'NO', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'CLOCK12', // 提示的名称，和value保持一致
      value: 'CLOCK12', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'CLOCK11', // 提示的名称，和value保持一致
      value: 'CLOCK11', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'CLOCK10', // 提示的名称，和value保持一致
      value: 'CLOCK10', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'CLOCK9', // 提示的名称，和value保持一致
      value: 'CLOCK9', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'CLOCK8', // 提示的名称，和value保持一致
      value: 'CLOCK8', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'CLOCK7', // 提示的名称，和value保持一致
      value: 'CLOCK7', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'CLOCK6', // 提示的名称，和value保持一致
      value: 'CLOCK6', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'CLOCK5', // 提示的名称，和value保持一致
      value: 'CLOCK5', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'CLOCK4', // 提示的名称，和value保持一致
      value: 'CLOCK4', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'CLOCK3', // 提示的名称，和value保持一致
      value: 'CLOCK3', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'CLOCK2', // 提示的名称，和value保持一致
      value: 'CLOCK2', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'CLOCK1', // 提示的名称，和value保持一致
      value: 'CLOCK1', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'ARROW_N', // 提示的名称，和value保持一致
      value: 'ARROW_N', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'ARROW_NE', // 提示的名称，和value保持一致
      value: 'ARROW_NE', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'ARROW_E', // 提示的名称，和value保持一致
      value: 'ARROW_E', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'ARROW_SE', // 提示的名称，和value保持一致
      value: 'ARROW_SE', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'ARROW_S', // 提示的名称，和value保持一致
      value: 'ARROW_S', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'ARROW_SW', // 提示的名称，和value保持一致
      value: 'ARROW_SW', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'ARROW_W', // 提示的名称，和value保持一致
      value: 'ARROW_W', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'ARROW_NW', // 提示的名称，和value保持一致
      value: 'ARROW_NW', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'TRIANGLE', // 提示的名称，和value保持一致
      value: 'TRIANGLE', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'TRIANGLE_LEFT', // 提示的名称，和value保持一致
      value: 'TRIANGLE_LEFT', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'CHESSBOARD', // 提示的名称，和value保持一致
      value: 'CHESSBOARD', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'DIAMOND', // 提示的名称，和value保持一致
      value: 'DIAMOND', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'DIAMOND_SMALL', // 提示的名称，和value保持一致
      value: 'DIAMOND_SMALL', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'SQUARE', // 提示的名称，和value保持一致
      value: 'SQUARE', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'SQUARE_SMALL', // 提示的名称，和value保持一致
      value: 'SQUARE_SMALL', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'RABBIT', // 提示的名称，和value保持一致
      value: 'RABBIT', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'COW', // 提示的名称，和value保持一致
      value: 'COW', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'MUSIC_CROTCHET', // 提示的名称，和value保持一致
      value: 'MUSIC_CROTCHET', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'MUSIC_QUAVER', // 提示的名称，和value保持一致
      value: 'MUSIC_QUAVER', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'MUSIC_QUAVERS', // 提示的名称，和value保持一致
      value: 'MUSIC_QUAVERS', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'PITCHFORK', // 提示的名称，和value保持一致
      value: 'PITCHFORK', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'XMAS', // 提示的名称，和value保持一致
      value: 'XMAS', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'PACMAN', // 提示的名称，和value保持一致
      value: 'PACMAN', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'TARGET', // 提示的名称，和value保持一致
      value: 'TARGET', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'TSHIRT', // 提示的名称，和value保持一致
      value: 'TSHIRT', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'ROLLERSKATE', // 提示的名称，和value保持一致
      value: 'ROLLERSKATE', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'DUCK', // 提示的名称，和value保持一致
      value: 'DUCK', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'HOUSE', // 提示的名称，和value保持一致
      value: 'HOUSE', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'TORTOISE', // 提示的名称，和value保持一致
      value: 'TORTOISE', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'BUTTERFLY', // 提示的名称，和value保持一致
      value: 'BUTTERFLY', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'STICKFIGURE', // 提示的名称，和value保持一致
      value: 'STICKFIGURE', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'GHOST', // 提示的名称，和value保持一致
      value: 'GHOST', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'SWORD', // 提示的名称，和value保持一致
      value: 'SWORD', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'GIRAFFE', // 提示的名称，和value保持一致
      value: 'GIRAFFE', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'SKULL', // 提示的名称，和value保持一致
      value: 'SKULL', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'SKULL', // 提示的名称，和value保持一致
      value: 'SKULL', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'SNAKE', // 提示的名称，和value保持一致
      value: 'SNAKE', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'ALL_CLOCKS', // 提示的名称，和value保持一致
      value: 'ALL_CLOCKS', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'ALL_ARROWS', // 提示的名称，和value保持一致
      value: 'ALL_ARROWS', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'uart', // 提示的名称，和value保持一致
    value: 'uart', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'init', // 提示的名称，和value保持一致
      value: 'init', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'read', // 提示的名称，和value保持一致
      value: 'read', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'any', // 提示的名称，和value保持一致
      value: 'any', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'readall', // 提示的名称，和value保持一致
      value: 'readall', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'readline', // 提示的名称，和value保持一致
      value: 'readline', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'readinto', // 提示的名称，和value保持一致
      value: 'readinto', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'write', // 提示的名称，和value保持一致
      value: 'write', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'spi', // 提示的名称，和value保持一致
    value: 'spi', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'init', // 提示的名称，和value保持一致
      value: 'init', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'read', // 提示的名称，和value保持一致
      value: 'read', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'write', // 提示的名称，和value保持一致
      value: 'write', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'write_readinto', // 提示的名称，和value保持一致
      value: 'write_readinto', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'music', // 提示的名称，和value保持一致
    value: 'music', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'set_tempo', // 提示的名称，和value保持一致
      value: 'set_tempo', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'pitch', // 提示的名称，和value保持一致
      value: 'pitch', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'play', // 提示的名称，和value保持一致
      value: 'play', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'get_tempo', // 提示的名称，和value保持一致
      value: 'get_tempo', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'stop', // 提示的名称，和value保持一致
      value: 'stop', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'reset', // 提示的名称，和value保持一致
      value: 'reset', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'DADADADUM', // 提示的名称，和value保持一致
      value: 'DADADADUM', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'ENTERTAINER', // 提示的名称，和value保持一致
      value: 'ENTERTAINER', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'PRELUDE', // 提示的名称，和value保持一致
      value: 'PRELUDE', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'ODE', // 提示的名称，和value保持一致
      value: 'ODE', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'NYAN', // 提示的名称，和value保持一致
      value: 'NYAN', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'RINGTONE', // 提示的名称，和value保持一致
      value: 'RINGTONE', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'FUNK', // 提示的名称，和value保持一致
      value: 'FUNK', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'BLUES', // 提示的名称，和value保持一致
      value: 'BLUES', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'BIRTHDAY', // 提示的名称，和value保持一致
      value: 'BIRTHDAY', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'WEDDING', // 提示的名称，和value保持一致
      value: 'WEDDING', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'FUNERAL', // 提示的名称，和value保持一致
      value: 'FUNERAL', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'PUNCHLINE', // 提示的名称，和value保持一致
      value: 'PUNCHLINE', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'PYTHON', // 提示的名称，和value保持一致
      value: 'PYTHON', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'BADDY', // 提示的名称，和value保持一致
      value: 'BADDY', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'CHASE', // 提示的名称，和value保持一致
      value: 'CHASE', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'BA_DING', // 提示的名称，和value保持一致
      value: 'BA_DING', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'WAWAWAWAA', // 提示的名称，和value保持一致
      value: 'WAWAWAWAA', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'JUMP_UP', // 提示的名称，和value保持一致
      value: 'JUMP_UP', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'JUMP_DOWN', // 提示的名称，和value保持一致
      value: 'JUMP_DOWN', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'POWER_UP', // 提示的名称，和value保持一致
      value: 'POWER_UP', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'POWER_DOWN', // 提示的名称，和value保持一致
      value: 'POWER_DOWN', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'antigravity', // 提示的名称，和value保持一致
    value: 'antigravity', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
  }, {
    name: 'this', // 提示的名称，和value保持一致
    value: 'this', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'authors', // 提示的名称，和value保持一致
      value: 'authors', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'love', // 提示的名称，和value保持一致
    value: 'love', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'badaboom', // 提示的名称，和value保持一致
      value: 'badaboom', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'neopixel', // 提示的名称，和value保持一致
    value: 'neopixel', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'NeoPixel', // 提示的名称，和value保持一致
      value: 'NeoPixel', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
      children: [{
        name: 'clear', // 提示的名称，和value保持一致
        value: 'clear', // 具体的提示代码
        score: 100, // 代码的排序
        meta: 'function', // 代码的分类,描述
      }, {
        name: 'show', // 提示的名称，和value保持一致
        value: 'show', // 具体的提示代码
        score: 100, // 代码的排序
        meta: 'function', // 代码的分类,描述
      }]
    }]
  }, {
    name: 'random', // 提示的名称，和value保持一致
    value: 'random', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'getrandbits', // 提示的名称，和value保持一致
      value: 'getrandbits', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'seed', // 提示的名称，和value保持一致
      value: 'seed', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'randint', // 提示的名称，和value保持一致
      value: 'randint', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'randrange', // 提示的名称，和value保持一致
      value: 'randrange', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'choice', // 提示的名称，和value保持一致
      value: 'choice', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'random', // 提示的名称，和value保持一致
      value: 'random', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'uniform', // 提示的名称，和value保持一致
      value: 'uniform', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'os', // 提示的名称，和value保持一致
    value: 'os', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'listdir', // 提示的名称，和value保持一致
      value: 'listdir', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'remove', // 提示的名称，和value保持一致
      value: 'remove', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'size', // 提示的名称，和value保持一致
      value: 'size', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'uname', // 提示的名称，和value保持一致
      value: 'uname', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'sys', // 提示的名称，和value保持一致
    value: 'sys', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'version', // 提示的名称，和value保持一致
      value: 'version', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'version_info', // 提示的名称，和value保持一致
      value: 'version_info', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'implementation', // 提示的名称，和value保持一致
      value: 'implementation', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'platform', // 提示的名称，和value保持一致
      value: 'platform', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'byteorder', // 提示的名称，和value保持一致
      value: 'byteorder', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'print_exception', // 提示的名称，和value保持一致
      value: 'print_exception', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'find', // 提示的名称，和value保持一致
    value: 'find', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'rfind', // 提示的名称，和value保持一致
    value: 'rfind', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'index', // 提示的名称，和value保持一致
    value: 'index', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'rindex', // 提示的名称，和value保持一致
    value: 'rindex', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'join', // 提示的名称，和value保持一致
    value: 'join', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'split', // 提示的名称，和value保持一致
    value: 'split', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'rsplit', // 提示的名称，和value保持一致
    value: 'rsplit', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'startswith', // 提示的名称，和value保持一致
    value: 'startswith', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'endswith', // 提示的名称，和value保持一致
    value: 'endswith', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'strip', // 提示的名称，和value保持一致
    value: 'strip', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'lstrip', // 提示的名称，和value保持一致
    value: 'lstrip', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'rstrip', // 提示的名称，和value保持一致
    value: 'rstrip', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'format', // 提示的名称，和value保持一致
    value: 'format', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'replace', // 提示的名称，和value保持一致
    value: 'replace', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'count', // 提示的名称，和value保持一致
    value: 'count', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'partition', // 提示的名称，和value保持一致
    value: 'partition', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'rpartition', // 提示的名称，和value保持一致
    value: 'rpartition', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'lower', // 提示的名称，和value保持一致
    value: 'lower', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'upper', // 提示的名称，和value保持一致
    value: 'upper', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'isspace', // 提示的名称，和value保持一致
    value: 'isspace', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'isalpha', // 提示的名称，和value保持一致
    value: 'isalpha', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'isdigit', // 提示的名称，和value保持一致
    value: 'isdigit', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'isupper', // 提示的名称，和value保持一致
    value: 'isupper', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'islower', // 提示的名称，和value保持一致
    value: 'islower', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'math', // 提示的名称，和value保持一致
    value: 'math', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
    children: [{
      name: 'sqrt', // 提示的名称，和value保持一致
      value: 'sqrt', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'pow', // 提示的名称，和value保持一致
      value: 'pow', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'exp', // 提示的名称，和value保持一致
      value: 'exp', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'log', // 提示的名称，和value保持一致
      value: 'log', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'cos', // 提示的名称，和value保持一致
      value: 'cos', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'sin', // 提示的名称，和value保持一致
      value: 'sin', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'tan', // 提示的名称，和value保持一致
      value: 'tan', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'acos', // 提示的名称，和value保持一致
      value: 'acos', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'asin', // 提示的名称，和value保持一致
      value: 'asin', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'atan', // 提示的名称，和value保持一致
      value: 'atan', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'atan2', // 提示的名称，和value保持一致
      value: 'atan2', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'ceil', // 提示的名称，和value保持一致
      value: 'ceil', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'copysign', // 提示的名称，和value保持一致
      value: 'copysign', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'fabs', // 提示的名称，和value保持一致
      value: 'fabs', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'floor', // 提示的名称，和value保持一致
      value: 'floor', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'fmod', // 提示的名称，和value保持一致
      value: 'fmod', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'frexp', // 提示的名称，和value保持一致
      value: 'frexp', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'ldexp', // 提示的名称，和value保持一致
      value: 'ldexp', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'modf', // 提示的名称，和value保持一致
      value: 'modf', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'isfinite', // 提示的名称，和value保持一致
      value: 'isfinite', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'isinf', // 提示的名称，和value保持一致
      value: 'isinf', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'isnan', // 提示的名称，和value保持一致
      value: 'isnan', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'trunc', // 提示的名称，和value保持一致
      value: 'trunc', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'radians', // 提示的名称，和value保持一致
      value: 'radians', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'degrees', // 提示的名称，和value保持一致
      value: 'degrees', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'abs', // 提示的名称，和value保持一致
    value: 'abs', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'all', // 提示的名称，和value保持一致
    value: 'all', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'any', // 提示的名称，和value保持一致
    value: 'any', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'bin', // 提示的名称，和value保持一致
    value: 'bin', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'bool', // 提示的名称，和value保持一致
    value: 'bool', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'bytearray', // 提示的名称，和value保持一致
    value: 'bytearray', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'bytes', // 提示的名称，和value保持一致
    value: 'bytes', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'callable', // 提示的名称，和value保持一致
    value: 'callable', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'chr', // 提示的名称，和value保持一致
    value: 'chr', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'classmethod', // 提示的名称，和value保持一致
    value: 'classmethod', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'dict', // 提示的名称，和value保持一致
    value: 'dict', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'dir', // 提示的名称，和value保持一致
    value: 'dir', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'divmod', // 提示的名称，和value保持一致
    value: 'divmod', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'enumerate', // 提示的名称，和value保持一致
    value: 'enumerate', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'eval', // 提示的名称，和value保持一致
    value: 'eval', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'exec', // 提示的名称，和value保持一致
    value: 'exec', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'filter', // 提示的名称，和value保持一致
    value: 'filter', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'float', // 提示的名称，和value保持一致
    value: 'float', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'frozenset', // 提示的名称，和value保持一致
    value: 'frozenset', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'getattr', // 提示的名称，和value保持一致
    value: 'getattr', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'globals', // 提示的名称，和value保持一致
    value: 'globals', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'hasattr', // 提示的名称，和value保持一致
    value: 'hasattr', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'hash', // 提示的名称，和value保持一致
    value: 'hash', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'help', // 提示的名称，和value保持一致
    value: 'help', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'hex', // 提示的名称，和value保持一致
    value: 'hex', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'id', // 提示的名称，和value保持一致
    value: 'id', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'int', // 提示的名称，和value保持一致
    value: 'int', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'isinstance', // 提示的名称，和value保持一致
    value: 'isinstance', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'issubclass', // 提示的名称，和value保持一致
    value: 'issubclass', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'iter', // 提示的名称，和value保持一致
    value: 'iter', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'len', // 提示的名称，和value保持一致
    value: 'len', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'list', // 提示的名称，和value保持一致
    value: 'list', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'locals', // 提示的名称，和value保持一致
    value: 'locals', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'map', // 提示的名称，和value保持一致
    value: 'map', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'max', // 提示的名称，和value保持一致
    value: 'max', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'min', // 提示的名称，和value保持一致
    value: 'min', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'next', // 提示的名称，和value保持一致
    value: 'next', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'object', // 提示的名称，和value保持一致
    value: 'object', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'oct', // 提示的名称，和value保持一致
    value: 'oct', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'open', // 提示的名称，和value保持一致
    value: 'open', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'ord', // 提示的名称，和value保持一致
    value: 'ord', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'pow', // 提示的名称，和value保持一致
    value: 'pow', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'range', // 提示的名称，和value保持一致
    value: 'range', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'repr', // 提示的名称，和value保持一致
    value: 'repr', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'reversed', // 提示的名称，和value保持一致
    value: 'reversed', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'round', // 提示的名称，和value保持一致
    value: 'round', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'set', // 提示的名称，和value保持一致
    value: 'set', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'setattr', // 提示的名称，和value保持一致
    value: 'setattr', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'sorted', // 提示的名称，和value保持一致
    value: 'sorted', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'staticmethod', // 提示的名称，和value保持一致
    value: 'staticmethod', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'str', // 提示的名称，和value保持一致
    value: 'str', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'sum', // 提示的名称，和value保持一致
    value: 'sum', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'super', // 提示的名称，和value保持一致
    value: 'super', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'tuple', // 提示的名称，和value保持一致
    value: 'tuple', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'type', // 提示的名称，和value保持一致
    value: 'type', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'zip', // 提示的名称，和value保持一致
    value: 'zip', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'function', // 代码的分类,描述
  }, {
    name: 'radio', // 提示的名称，和value保持一致
    value: 'radio', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'on', // 提示的名称，和value保持一致
      value: 'on', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'off', // 提示的名称，和value保持一致
      value: 'off', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'config', // 提示的名称，和value保持一致
      value: 'config', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'reset', // 提示的名称，和value保持一致
      value: 'reset', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'send_bytes', // 提示的名称，和value保持一致
      value: 'send_bytes', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'receive_bytes', // 提示的名称，和value保持一致
      value: 'receive_bytes', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'send', // 提示的名称，和value保持一致
      value: 'send', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'RATE_250KBIT', // 提示的名称，和value保持一致
      value: 'RATE_250KBIT', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'RATE_1MBIT', // 提示的名称，和value保持一致
      value: 'RATE_1MBIT', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }, {
      name: 'RATE_2MBIT', // 提示的名称，和value保持一致
      value: 'RATE_2MBIT', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述
    }]
  }, {
    name: 'audio', // 提示的名称，和value保持一致
    value: 'audio', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'play', // 提示的名称，和value保持一致
      value: 'play', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述        
    }, {
      name: 'AudioFrame', // 提示的名称，和value保持一致
      value: 'AudioFrame', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述        
    }]
  }, {
    name: 'speech', // 提示的名称，和value保持一致
    value: 'speech', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'translate', // 提示的名称，和value保持一致
      value: 'translate', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述        
    }, {
      name: 'say', // 提示的名称，和value保持一致
      value: 'say', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述        
    }, {
      name: 'pronounce', // 提示的名称，和value保持一致
      value: 'pronounce', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述        
    }, {
      name: 'sing', // 提示的名称，和value保持一致
      value: 'sing', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述        
    }]
  }, {
    name: 'Obloq', // 提示的名称，和value保持一致
    value: 'Obloq', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'connectWifi', // 提示的名称，和value保持一致
      value: 'connectWifi', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述        
    }, {
      name: 'ifconfig', // 提示的名称，和value保持一致
      value: 'ifconfig', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述        
    }, {
      name: 'httpConfig', // 提示的名称，和value保持一致
      value: 'httpConfig', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述        
    }, {
      name: 'get', // 提示的名称，和value保持一致
      value: 'get', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述        
    }, {
      name: 'post', // 提示的名称，和value保持一致
      value: 'post', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述        
    }, {
      name: 'mqttConfig', // 提示的名称，和value保持一致
      value: 'mqttConfig', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述        
    }, {
      name: 'connectMqtt', // 提示的名称，和value保持一致
      value: 'connectMqtt', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述        
    }, {
      name: 'subscribe', // 提示的名称，和value保持一致
      value: 'subscribe', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述        
    }, {
      name: 'publish', // 提示的名称，和value保持一致
      value: 'publish', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述        
    }]
  }, {
    name: 'RFID', // 提示的名称，和value保持一致
    value: 'RFID', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述
    children: [{
      name: 'read', // 提示的名称，和value保持一致
      value: 'read', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述        
    }, {
      name: 'stopAutoRead', // 提示的名称，和value保持一致
      value: 'stopAutoRead', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述        
    }, {
      name: 'write', // 提示的名称，和value保持一致
      value: 'write', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述        
    }, {
      name: 'buzzerOpen', // 提示的名称，和value保持一致
      value: 'buzzerOpen', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述        
    }, {
      name: 'buzzerClose', // 提示的名称，和value保持一致
      value: 'buzzerClose', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述        
    }, {
      name: 'reset', // 提示的名称，和value保持一致
      value: 'reset', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述        
    }, {
      name: 'recognize', // 提示的名称，和value保持一致
      value: 'recognize', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述        
    }]
  }, {
    name: 'DFMotor', // 提示的名称，和value保持一致
    value: 'DFMotor', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述      
    children: [{
      name: 'speed', // 提示的名称，和value保持一致
      value: 'speed', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述  
    }, {
      name: 'run', // 提示的名称，和value保持一致
      value: 'run', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述  
    }, {
      name: 'stop', // 提示的名称，和value保持一致
      value: 'stop', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述  
    }]
  }, {
    name: 'DFServo', // 提示的名称，和value保持一致
    value: 'DFServo', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述      
    children: [{
      name: 'angle', // 提示的名称，和value保持一致
      value: 'angle', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述  
    }]
  }, {
    name: 'DFStepper', // 提示的名称，和value保持一致
    value: 'DFStepper', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述      
    children: [{
      name: 'angle', // 提示的名称，和value保持一致
      value: 'angle', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述  
    }, {
      name: 'circle', // 提示的名称，和value保持一致
      value: 'circle', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述  
    }, {
      name: 'stop', // 提示的名称，和value保持一致
      value: 'stop', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述  
    }]
  }, {
    name: 'Servo', // 提示的名称，和value保持一致
    value: 'Servo', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述      
    children: [{
      name: 'angle', // 提示的名称，和value保持一致
      value: 'angle', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述  
    }, {
      name: 'read', // 提示的名称，和value保持一致
      value: 'read', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述  
    }]
  }, {
    name: 'urm10', // 提示的名称，和value保持一致
    value: 'urm10', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述      
    children: [{
      name: 'read', // 提示的名称，和value保持一致
      value: 'read', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述  
    }]
  }, {
    name: 'necir', // 提示的名称，和value保持一致
    value: 'necir', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述      
    children: [{
      name: 'read', // 提示的名称，和value保持一致
      value: 'read', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述  
    }]
  }, {
    name: 'dht11', // 提示的名称，和value保持一致
    value: 'dht11', // 具体的提示代码
    score: 100, // 代码的排序
    meta: 'object', // 代码的分类,描述      
    children: [{
      name: 'read', // 提示的名称，和value保持一致
      value: 'read', // 具体的提示代码
      score: 100, // 代码的排序
      meta: 'function', // 代码的分类,描述  
    }]
  }


]

const pyCodeCompleter = {
  getCompletions: function(editor, session, pos, prefix, callback) {
    const completions = [];
    pythonCode.forEach((code) => {
      completions.push({
        caption: code.name,
        value: code.value,
        score: code.score,
        meta: code.meta,
        children: code.children || ''
      });
    });
    callback(null, completions);
  }
};

export default pyCodeCompleter;
